apollo-api/src/main/java/com/apollographql/android/api/graphql/Field.java

```diff
@@ -89,7 +96,64 @@ public boolean optional() {
     return optional;
   }
 
-  public static enum Type {
+  public String cacheKey(Operation.Variables variables) {
+    if (arguments.isEmpty()) {
+      return fieldName();
+    }
+    return String.format("%s(%s)", fieldName(), orderIndependentKey(arguments, variables));
+  }
+
+  private String orderIndependentKey(Map<String, Object> objectMap, Operation.Variables variables) {
```

Comment:
👍 I'll pull out the key for variable as it is conceptually different from the rest